lib/Sema/SemaType.cpp

```diff
@@ -2014,6 +2014,13 @@ QualType Sema::BuildPointerType(QualType T, CheckedPointerKind kind,
     return QualType();
   }
 
+  // In Checked C, _MMSafe_ptr is only allowed to point to struct types.
+  if (kind == CheckedPointerKind::MMSafe&& !T->isStructureType()) {
```

Comment:
This is a bug here. Currently this code would fail to compile when we have a `_MMSafe_ptr<T> p` in a generic function.
The compiler complains that 'T' is not a struct type. I'll fix this soon.

Reply:
Space needed between CheckedPointerKind::MMSafe&&

Reply:
Thanks. I noticed that right after the commit. I'll fix it in next commit.